doc/man1/flux-uri.rst

```diff
@@ -0,0 +1,144 @@
+is assumed to be ``jobid``.  This allows ``flux uri`` to be used to look
+up the URI for a Flux instance running as a job with:
+
+::
+
+   $ flux uri JOBID
+
+Depending on the *TARGET* URI scheme and corresponding plugin, specific
+query arguments int *TARGET* may be supported. However, as a convenience,
+all *TARGET* URIs support the special query arguments ``local`` or
+``remote`` to force the resulting URI into a local (``local://``) or remote
+(``ssh://``) form. For example:
+
+::
+
+   $ flux uri JOBID?local
+
+would return the ``local://`` URI for *JOBID* (if the URI can be resolved).
+
+A list of supported URI schemes will be listed at the bottom of
+``flux uri --help`` message. For a description of the URI resolver schemes
+included with Flux, see the URI SCHEMES and EXAMPLES sections below.
+
+OPTIONS
+=======
+
+**-remote**
+   Return the _remote_ (``ssh://``)  equivalent of the resolved URI.
+
+**--local**
+   Return the _local_ (``local://``) equivalent of the resulved URI.
+   Warning: the resulting URI may be invalid for the current system
+   if the network host specified by an ``ssh`` URI is not the current
+   host.
+
+URI SCHEMES
+===========
+
+The following URI schemes are included by default:
+
+jobid:ID[/ID...]
+   This scheme attempts to get the URI for a Flux instance running as a
+   job in the current enclosing instance. This is the assumed scheme if no
+   ``scheme:`` is provided in *TARGET* passed to ``flux uri``, so the
+   ``jobid:`` prefix is optional. A hierarchy of Flux jobids is supported,
+   so ``f1234/f3456`` will resolve the URI for job ``f3456`` running in
+   job ``f1234`` in the current instance.
+
+pid:PID
+  This scheme attempts to read the ``FLUX_URI`` value from the process id
+  *PID* using ``/proc/PID/environ``. If *PID* refers to a ``flux-broker``,
+  then a child of the broker is used instead in order to obtain the
+  URI for that broker (instead of its parent)
+
+slurm:JOBID
+  This scheme makes a best-effort to resolve the URI of a Flux instance
+  launched under Slurm. It invokes ``srun`` to run ``scontrol listpids``
+  on the first node of the job, and then uses the ``pid`` resolver until
+  it finds a valid ``FLUX_URI``.
+
+
+EXAMPLES
+========
+
+To get the URI of a job in the current instance in its ``local://`` form:
+
+::
+
+   $ flux uri --local ƒN8Pz2xVu
+   local:///tmp/flux-zbVtVg/jobtmp-0-ƒN8Pz2xVu/flux-59uf5w/local-0
+
+or
+
+::
+
+   $ flux uri ƒN8Pz2xVu?local
+   local:///tmp/flux-zbVtVg/jobtmp-0-ƒN8Pz2xVu/flux-59uf5w/local-0
+
+
+Get the URI of a nested job:
+
+::
+
+   $ flux uri ƒqxxTiZBM/ƒr2XFWP?local
+   local:///tmp/flux-zbVtVg/jobtmp-0-ƒqxxTiZBM/flux-EPgSwk/local-0
+
+.. note::
+   With  the ``jobid`` resolver, ``?local`` only needs to be placed on
+   the last component of the jobid "path" or hierarchy. This will resolve
+   each URI in turn as a local URI, so it is only useful if all jobs
+   are actually running on the local system.
```

Comment:
If the last job is running on the local system, then all enclosing instances have to be running on the local system, so maybe the "so..." explanation could just be dropped?

Reply:
Yeah, that sounds better, though more of what I was going for is that in resolving each job along the way the jobid resolver plugin has to connect to the parent job to resolve its child, and when `?local` is placed on the last jobid _all_ connections will use the `local` connector. This may not work if for example the first jobid is on another host (the first `flux_open()` will fail).

Also, jobid URIs are resolved to the URI of rank 0. If not all enclosing instances have rank 0 running locally, then your statement, while true, doesn't apply to resolution of URIs.

However, perhaps that is too much detail for the manpage.

Reply:
Up to you, but if you leave it in you might want to include the rank 0 requirement.

> If not all enclosing instances have rank 0 running locally

The top enclosing instance wouldn't have to be local, correct?

Reply:
No, I just left it out.

> The top enclosing instance rank 0 isn't required to be local, correct?

no I meant the first jobid in the path component, it will be forced to a `local://` uri if `?local` is used. Probably the only use case for `?local` is developers working with test instances, so it is probably better to just sweep this under the rug for now.